src/components/Settings/SettingsAddressbookShare.vue

```diff
@@ -85,32 +81,94 @@ export default {
 	data() {
 		return {
 			isLoading: false,
-			usersOrGroups: [],
-			selectedUserOrGroup: []
+			usersOrGroups: []
+		}
+	},
+	computed: {
+		placeholder() {
+			return t('contacts', 'Share with users or groups')
+		},
+		noResult() {
+			return t('contacts', 'Oops! No elements found. Consider changing the search query.')
 		}
 	},
 	methods: {
-		limitText(count) {
-			return `and ${count} other users or groups`
+		/**
+		 * Share addressbook
+		 *
+		 * @param {Object} chosenUserOrGroup
+		 */
+		shareAddressbook(chosenUserOrGroup) {
```

Comment:
you can also replace `chosenUserOrGroup` by `{ sharee, group }`
Easier to understand and you don't need the `let xxx = `

Reply:
I am not sure how this should work, the chosenUserOrGroup is an object that is the return from the custom options, it looks like this:

let newMatch = {
					match: matches[i],
					matchstart: matchResult[0],
					matchpattern: matches[i].match(regex)[0],
					matchend: matchResult[1],
					matchtag: group ? '(group)' : '(user)',
					matchgroup: group
				}

How do I tell the multiselect to only pass on the match and matchgroup as suggested?

Reply:
You can still pass the full object and destructure the only desired props :)

``` js
({ match, matchgroup } = {
    match: 0,
    matchstart: 1,
    matchpattern: 2,
    matchend: 3,
    matchtag: '(user)',
    matchgroup: 4
})

match
> 0
matchgroup
> 4
```

Reply:
💯